### PR TITLE
[DESK-771][DESK-777] account linking blank screen

### PR DESCRIPTION
**Pull request by JamieRuderman**
### Description

device list sharing
registration workflow
refreshing single devices
viewing shared device list connections
Fix for graphQL error when device isn't registered

### Test plan

1. Sign in
2. Go to...
3. Click...
4. You should see X, Y, Z...

### Ticket

<https://remoteit.atlassian.net/browse/DESK-771>
<https://remoteit.atlassian.net/browse/DESK-777>

### CODE REVIEW

File: frontend/src/helpers/connectionHelper.ts [hunk 1]
```diff
@@ -49,6 +51,13 @@ export function clearConnectionError(connection: IConnection) {
   emit('connection', connection)
 }
 
+export function getConnectionIds(state: ApplicationState) {
+  const { device, connections } = state.backend
+  let ids = connections.map(c => c.id)
```
Comment (mauricio0000) [inline frontend/src/helpers/connectionHelper.ts:56]: why do you need a `let`?

File: frontend/src/helpers/connectionHelper.ts [hunk 2]
```diff
@@ -49,6 +51,13 @@ export function clearConnectionError(connection: IConnection) {
...
+  if (device.uid) ids.push(device.uid)
```
Comment (mauricio0000) [inline frontend/src/helpers/connectionHelper.ts:57]: could you add validation to check if the id is not duplicate? I am not sure

File: backend/src/Controller.ts [hunk 3]
```diff
@@ -161,8 +160,7 @@ class Controller {
 
   clearAll = async () => {
     Logger.info('CLEAR CREDENTIALS')
-    await user.clearAll() // moved to before clearing connections because having many connection would cause sign out to fail.
-    await this.pool.clearAll()
+    await user.clearAll()
```
Comment (mauricio0000) [inline backend/src/Controller.ts:163]: why do you need `await`?

File: frontend/src/models/devices.ts [hunk 4]
```diff
@@ -97,38 +92,37 @@ export default createModel({
...
       try {
-        const gqlResponse = await graphQLFetchDevice(deviceId, accountId)
+        const gqlResponse = await graphQLFetchDevice(deviceId)
         graphQLGetErrors(gqlResponse)
-        const { device } = gqlResponse?.data?.data?.login?.account || {}
-        result = graphQLAdaptor([device], accountId)[0]
+        const { device } = gqlResponse?.data?.data?.login || {}
+        const loginId = gqlResponse?.data?.data?.login?.id
+        result = device ? graphQLAdaptor(device, loginId, hidden)[0] : []
+        debugger
```
Comment (mauricio0000) [inline frontend/src/models/devices.ts:109]: could you remove the `debugger`?

**Review by mauricio0000 — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/connectionHelper.ts (hunk 1 above)
Comment (JamieRuderman) [reply frontend/src/helpers/connectionHelper.ts:56]: well I'm pushing to the array in the next line, so I assume it's gotta be let or var... not const if it's changing.  🤷‍♂️

File: frontend/src/helpers/connectionHelper.ts (hunk 2 above)
Comment (JamieRuderman) [reply frontend/src/helpers/connectionHelper.ts:57]: Sure why not!

File: backend/src/Controller.ts (hunk 3 above)
Comment (JamieRuderman) [reply backend/src/Controller.ts:163]: No, don't need await... and this is actually changing again... I'll update...

File: frontend/src/models/devices.ts (hunk 4 above)
Comment (JamieRuderman) [reply frontend/src/models/devices.ts:109]: Yes. Definitely.

**Review by mauricio0000 — APPROVED** [review]

**Pull request merged**